tss-esapi/src/context/tpm_commands/asymmetric_primitives.rs

```diff
@@ -43,6 +164,125 @@ impl Context {
     }
 
     /// Perform an asymmetric RSA decryption.
+    ///
+    /// # Details
+    /// *From the specification*
+    /// > This command performs RSA decryption using the indicated padding scheme according to IETF RFC
+    /// > 8017 ((PKCS#1).
+    ///
+    /// # Arguments
+    /// * `key_handle`  - A [KeyHandle] of the RSA key to use for decryption.
+    /// * `cipher_text` - The cipher text to be decrypted.
+    /// * `in_scheme`  - The padding scheme to use if scheme associated with
+    ///   the `key_handle` is [RsaDecryptionScheme::Null].
+    /// * `label`       - A label whose association with the message is to be verified.
```

Comment:
Does the label have to match the label used during encrypt?

Reply:
*TCG TPM 2.0 Specification, Revision 01.59, Part 3 - Commands, Section 14.3 - TPM2_RSA_Decrypt*
> If a label is used in the padding process of the scheme during encryption, the label parameter is required
> to be present in the decryption process and label is required to be the same in both cases. If label is not
> the same, the decrypt operation is very likely to fail ((TPM_RC_VALUE). If label is present (label.size !=
> 0), it shall be a byte stream whose last byte is zero or the TPM will return TPM_RC_VALUE.
> NOTE 1 The size of label includes the terminating null.

I also noticed another thing that the label is optional. So I will have to adress that as well.